### DISCUSSION TITLE
Install without `nix`

### DISCUSSION THREAD

**Discussion by notramo** [srid/emanote#536] [Ideas]
Not everyone uses NixOS, and I don't want to complicate my lean and simple Void Linux install with a bulky build system that is completely different from the principles of Void Linux.
Surely, Nix is cool and has its uses, but unfortunately it does not fit every system.

Is it possible to use without Nix?
Or at least could you provide binary builds in the GitHub releases? (If it's possible to run with only a simple binary file.)

**Comment by srid** [comment]
If someone provides a Github Workflow for building static binaries (see https://github.com/nh2/static-haskell-nix), I'd be happy to accept that contribution and have it be part of the GitHub releases.

**Reply by notramo** [reply]
Sadly, I'm not very keen on contributing to software that I can't even try.

**Reply by vcavallo** [reply]
how would cross-platform integrations ever happen if people didn't contribute to software that "they couldn't (previously) try"?

**Comment by zoranbosnjak** [comment]
@srid 
Somewhat related to the original discussion..., I would like to know if there is a way to install `emanote` with plain `nix`, without `nix flakes`? I am already using `nix` package manager on top of ubuntu os. So far, I have survived without `nix flakes`, but the `emanote` install instructions suggest to use it. So I have added the scary `'--extra-experimental-features nix-command flakes'` to the command line as suggested by the prompt. The install was successfull, but now `nix` won't let me use old commands. `nix-env -i ...`, it says:

```
... the profile ... is incompatible with 'nix-env'; please use 'nix profile' instead 
``` 

I did not plan to switch to flakes permanently. I know it has nothing to do with emanote. Any other flakes project would get me to the same profile state. I would just like to check if there is an easy way to use the project with plain `nix` and how to switch back? Or maybe get your nix flakes opinion why not to switch back.

**Reply by srid** [reply]
You should be able to use it without flakes, via this repo's [`default.nix`](https://github.com/srid/emanote/blob/master/default.nix).

**Reply by srid** [reply]
https://nixos.wiki/wiki/Flakes#Using_flakes_with_stable_Nix

**Reply by srid** [reply]
Not related to this, but generally I do recommend people to have Flakes enabled via using the unofficial installer: https://nixos.asia/en/install

**Comment by nramirezuy** [comment]
@notramo do you care about using Docker?

[Dockerfile](https://github.com/nramirezuy/dockerfiles/blob/main/emanote/Dockerfile)
[Dockerhub](https://hub.docker.com/r/nramirezuy/emanote)
[GHCR](https://github.com/nramirezuy/dockerfiles/pkgs/container/emanote)

**Reply by srid** [reply]
Relate https://github.com/srid/emanote/pull/561
